patches/server/0740-Fire-EntityChangeBlockEvent-in-more-places.patch

```diff
@@ -6,6 +6,32 @@ Subject: [PATCH] Fire EntityChangeBlockEvent in more places
 Co-authored-by: ChristopheG <61288881+chrisgdt@users.noreply.github.com>
 Co-authored-by: maxcom1 <46265094+maxcom1@users.noreply.github.com>
 
+diff --git a/src/main/java/net/minecraft/world/effect/WeavingMobEffect.java b/src/main/java/net/minecraft/world/effect/WeavingMobEffect.java
+index 6a84eaeddd0d7f050053c8aa0659d6811192aad4..785188e78efaab4ca2f516da648c24e35a187f94 100644
+--- a/src/main/java/net/minecraft/world/effect/WeavingMobEffect.java
++++ b/src/main/java/net/minecraft/world/effect/WeavingMobEffect.java
+@@ -25,11 +25,11 @@ class WeavingMobEffect extends MobEffect {
+     @Override
+     public void onMobRemoved(LivingEntity entity, int amplifier, Entity.RemovalReason reason) {
+         if (reason == Entity.RemovalReason.KILLED && (entity instanceof Player || entity.level().getGameRules().getBoolean(GameRules.RULE_MOBGRIEFING))) {
+-            this.spawnCobwebsRandomlyAround(entity.level(), entity.getRandom(), entity.getOnPos());
++            this.spawnCobwebsRandomlyAround(entity, entity.level(), entity.getRandom(), entity.getOnPos()); // Paper - Fire EntityChangeBlockEvent in more places
+         }
+     }
+ 
+-    private void spawnCobwebsRandomlyAround(Level world, RandomSource random, BlockPos pos) {
++    private void spawnCobwebsRandomlyAround(LivingEntity entity, Level world, RandomSource random, BlockPos pos) { // Paper - Fire EntityChangeBlockEvent in more places
+         Set<BlockPos> set = Sets.newHashSet();
+         int i = this.maxCobwebs.applyAsInt(random);
+ 
+@@ -46,6 +46,7 @@ class WeavingMobEffect extends MobEffect {
+         }
+ 
+         for (BlockPos blockPos3 : set) {
++            if (!org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(entity, blockPos3, Blocks.COBWEB.defaultBlockState())) break; // Paper - Fire EntityChangeBlockEvent in more places
```

Comment:
Shouldn't it continue instead of breaking?

Reply:
oops yeah. 1 sec